e2e/utils.go

```diff
@@ -473,16 +473,128 @@ func (s SimplyBlock) getStoragenode() (string, error) {
 
 	// TODO: get a random storage node
 	storageNodes, ok := out.([]interface{})[0].(map[string]interface{})
+
 	if !ok {
 		return "", errors.New("failed to get storage node from simplyblock api")
 	}
 	sn, ok := storageNodes["hostname"].(string)
+	fmt.Println("^^^^^^")
+	fmt.Println(sn)
+	fmt.Println("^^^^^^")
 	if !ok {
 		return "", errors.New("failed to get storage node from simplyblock api")
 	}
 	return sn, nil
 }
 
+func (s SimplyBlock) numberOfNodes() (int, error) {
+	var rpcClient util.RPCClient
+	rpcClient.ClusterID = s.UUID
+	rpcClient.ClusterIP = s.IP
+	rpcClient.ClusterSecret = s.Secret
+
+	rpcClient.HTTPClient = &http.Client{Timeout: 10 * time.Second}
+
+	out, err := rpcClient.CallSBCLI("GET", "/storagenode", nil)
+	if err != nil {
+		return 0, err
+	}
+
+	//get the number of storage nodes
+	sn := len(out.([]interface{}))
+
+	return sn, nil
+
+}
+func (s SimplyBlock) getStoragenodeId(random int) (string, error) {
+	var rpcClient util.RPCClient
+	rpcClient.ClusterID = s.UUID
+	rpcClient.ClusterIP = s.IP
+	rpcClient.ClusterSecret = s.Secret
+
+	rpcClient.HTTPClient = &http.Client{Timeout: 10 * time.Second}
+
+	// get the list of storage nodes
+	out, err := rpcClient.CallSBCLI("GET", "/storagenode", nil)
+	if err != nil {
+		return "", err
+	}
+
+	// TODO: get a random storage node
+
+	storageNodes, ok := out.([]interface{})[random].(map[string]interface{})
+
+	if !ok {
+		return "", errors.New("failed to get storage node from simplyblock api")
+	}
+	sn, ok := storageNodes["id"].(string)
+	if !ok {
+		return "", errors.New("failed to get storage node from simplyblock api")
+	}
+	return sn, nil
+}
+
+func (s SimplyBlock) restartStorageNode(nodeID string) error {
+	var rpcClient util.RPCClient
+	rpcClient.ClusterID = s.UUID
+	rpcClient.ClusterIP = s.IP
+	rpcClient.ClusterSecret = s.Secret
+	rpcClient.HTTPClient = http.DefaultClient
+
+	// Step 1: Suspend Storage Node
+	url := fmt.Sprintf("/storagenode/suspend/%s", nodeID)
+	if _, err := rpcClient.CallSBCLI("GET", url, nil); err != nil {
+		return fmt.Errorf("failed to suspend storage node: %w", err)
+	}
+
+	// Step 2: Shutdown Storage Node
+	url = fmt.Sprintf("/storagenode/shutdown/%s/?force=True", nodeID)
```

Comment:
confirm the status of the node before proceeding to the next step

Reply:
Please check the storage node status

Reply:
sorry, can you please clarify, I am already checking the storage node status with status_code which is mapped to different statuses.

Reply:
I prefer we check using `status` that is string instead of `status_code`
The storage node object has `status` as it has `status_code`